localstack/utils/aws/arns.py

```diff
@@ -346,9 +346,9 @@ def kinesis_stream_name(kinesis_arn):
     return kinesis_arn.split(":stream/")[-1]
 
 
-def apigateway_invocations_arn(lambda_uri, region_name: str = None):
+def apigateway_invocations_arn(lambda_uri: str, region_name: str) -> str:
     return "arn:aws:apigateway:%s:lambda:path/2015-03-31/functions/%s/invocations" % (
-        region_name or get_region(),
+        region_name,
         lambda_uri,
     )
```

Comment:
q: should this use the region of the original lambda ARN instead? what is the purpose of overriding it?

Reply:
My knowledge of APIGW is non-existent, but it could be possible that the APIGW is invoking a Lambda function in another region?

Reply:
I think the invocation path should always the extended lambda ARN, basically if you create a lambda in `us-west-1` but modify the region to be `us-east-1` then the resource wouldn't exist, I believe APIGW will use the region of the invocation ARN to then invoke the lambda. It should be the region of the original lambda ARN.

Reply:
Yes, exactly, you're right! https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-cross-account-lambda-integrations.html

This is here. So yeah, we will need the region name for the APIGW region, this is right! Sorry for the misleading comment 😬

Reply:
This is the code which get the lambda ARN from the invocation ARN:
```python
# localstack.services.apigateway.integration.LambdaIntegration._lambda_integration_uri
uri = uri.split(":lambda:path")[1].split("functions/")[1].split("/invocations")[0]
```
So it does not take into account the APIGW region,

Reply:
I'll add to this: it seems we can indeed construct the ARN, but then AWS will reject it, it's not possible to do so. https://github.com/aws/aws-cdk/issues/17458

We should properly construct the wrong ARN anyway using the API region basically. But we should then reject the request if the invocation ARN passed to `create_integration` has 2 different regions in it, which is a different issue all together. Thanks for putting this into light, multi region is always showing nice behaviours we were not aware!